### ISSUE TITLE
Can support for BibDesk be added to bibtex-completion?

### ISSUE THREAD

**Issue by rhstanton** [tmalsburg/helm-bibtex#396]
I, like many other Mac users, use BibDesk to manage my bibliographies, and it would be very nice if bibtex-completion could find the PDF file for a given BibTeX entry using the bdsk-file-1 (or ...2 if there's more than one PDF file) entry that BibDesk puts into the bib file (which is a base64-encoded fully qualified path name).

This discussion came up a few years relative to org-ref. Below is a link to the discussion, which contains a solution that should be relatively easy to port over to bibtex-completion. Thanks!

[https://github.com/jkitchin/org-ref/issues/273](url)

**Comment by rhstanton** [comment]
Hmm. The link above doesn't seem to work for me if I click on it, but if I cut and paste the text into a browser it works fine.

**Comment by tmalsburg** [comment]
Hi, thanks for creating this issue.  In principle I would like to support all the different ways to reference PDFs.  But some bibliography managers make this more difficult than necessary, some likely intentionally to lock in users.  Although that's probably not the case with BibDesk which is libre software.

Anyway, the solution is to create a plugin system for Bibtex completion where the user configures the linking style used in their bibliography and then the appropriate plugin is used (could be multiple plugins for messy bibliographies).  We can't try all linking styles because there are too many and it would slow down parsing too much.

If anyone is interested in working on such a plug-in system, I would be happy to provide guidance.  It's actually not too difficult and such a system would close a lot of other issues as well.  I would like to work on this myself, but I probably won't find the time in the near future.

**Comment by rhstanton** [comment]
It looks like we could almost do it by just setting bibtex-completion-pdf-field to "bdsk-file-1”. However, the field contents need to be base64 decoded to obtain the file name + directory.

An example is 

bdsk-file-1 = {YnBsaXN0MDDSAQIDBFxyZWxhdGl2ZVBhdGhZYWxpYXNEYXRhXxAuLi4vLi4vLi4vLi4vLi4vRG9jdW1lbnRzL1BhcGVycy9BZGFtcy8xOTk0LnBkZk8RAUoAAAAAAUoAAgAADE1hY2ludG9zaCBIRAAAAAAAAAAAAAAAAAAAAAAAAABCRAAB/////wgxOTk0LnBkZgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAD/////AAAAAAAAAAAAAAAAAAUABAAACiBjdQAAAAAAAAAAAAAAAAAFQWRhbXMAAAIALy86VXNlcnM6c3RhbnRvbjpEb2N1bWVudHM6UGFwZXJzOkFkYW1zOjE5OTQucGRmAAAOABIACAAxADkAOQA0AC4AcABkAGYADwAaAAwATQBhAGMAaQBuAHQAbwBzAGgAIABIAEQAEgAtVXNlcnMvc3RhbnRvbi9Eb2N1bWVudHMvUGFwZXJzL0FkYW1zLzE5OTQucGRmAAATAAEvAAAVAAIADv//AAAACAANABoAJABVAAAAAAAAAgEAAAAAAAAABQAAAAAAAAAAAAAAAAAAAaM=}

If I put the contents of the field into file test.txt, then run

 base64 -d test.txt | plutil -convert xml1 -o - -

I get the output

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aliasData</key>
	<data>
	AAAAAAFKAAIAAAxNYWNpbnRvc2ggSEQAAAAAAAAAAAAAAAAAAAAAAAAAQkQAAf////8I
	MTk5NC5wZGYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
	AAAAAAAAAAAAAAAA/////wAAAAAAAAAAAAAAAAAFAAQAAAogY3UAAAAAAAAAAAAAAAAA
	BUFkYW1zAAACAC8vOlVzZXJzOnN0YW50b246RG9jdW1lbnRzOlBhcGVyczpBZGFtczox
	OTk0LnBkZgAADgASAAgAMQA5ADkANAAuAHAAZABmAA8AGgAMAE0AYQBjAGkAbgB0AG8A
	cwBoACAASABEABIALVVzZXJzL3N0YW50b24vRG9jdW1lbnRzL1BhcGVycy9BZGFtcy8x
	OTk0LnBkZgAAEwABLwAAFQACAA7//wAA
	</data>
	<key>relativePath</key>
	<string>../../../../../Documents/Papers/Adams/1994.pdf</string>
</dict>
</plist>

**Comment by tmalsburg** [comment]
Yes, it's not super difficult to implement. But the current approach is to try all known formats (bare path, Mendeley/Zotero format). If we do this with the present format, we'd have to check every single entry even for people who are not even using BibDesk. Too expensive.

**Comment by rhstanton** [comment]
Agreed. There should at least be a user-defined variable to tell bibtex-completion what to look for.
